### ISSUE TITLE
zstreamdump not available for in-tree testing

### ISSUE THREAD

**Issue by freqlabs** [openzfs/zfs#12211]
### System information

Type | Version/Name
 --- | ---
OS Name	| FreeBSD
OS Version	| 13, 14
Architecture	| amd64
ZFS Version	| master (08cd0717359b1a18693e3c8e6d6e5a2819b35a48)
SPL Version	| master (08cd0717359b1a18693e3c8e6d6e5a2819b35a48)

### Describe the problem you're observing
After dd00925e8 we are no longer creating a link for zstreamdump in bin/ for running ZTS in tree (./scripts/zfs-tests.sh).
This causes a few tests to fail and at least one loops indefinitely.

### Describe how to reproduce the problem
`./scripts/zfs-tests.sh -vx` lists missing programs, among them zstreamdump.

### Include any warning/errors/backtraces from the system logs

```
FreeBSD-14_0-CURRENT-3bede2908ac ➜  zfs git:(master) ./scripts/zfs-tests.sh -vxT redacted_send

--- Cleanup ---
Removing pool(s):
Removing loopback(s): md0 md1
Removing files(s):

--- Configuration ---
Runfiles:        /home/ryan/zfs/tests/runfiles/common.run,/home/ryan/zfs/tests/runfiles/freebsd.run
STF_TOOLS:       /home/ryan/zfs/tests/test-runner
STF_SUITE:       /home/ryan/zfs/tests/zfs-tests
STF_PATH:        /home/ryan/zfs/bin
FILEDIR:         /var/tmp
FILES:           /var/tmp/file-vdev0 /var/tmp/file-vdev1 /var/tmp/file-vdev2
LOOPBACKS:       md0 md1 md2
DISKS:           md0 md1 md2
NUM_DISKS:       3
FILESIZE:        4G
ITERATIONS:      1
TAGS:            redacted_send
STACK_TRACER:    no
Keep pool(s):    rpool
Missing util(s): mount.zfs zed zgenhostid zstreamdump devname2devid mmap_libaio randfree_file user_ns_exec xattrtest python2

/home/ryan/zfs/tests/test-runner/bin/test-runner.py  -c "/home/ryan/zfs/tests/runfiles/common.run,/home/ryan/zfs/tests/runfiles/freebsd.run" -T "redacted_send" -i "/home/ryan/zfs/tests/zfs-tests" -I "1"
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/setup (run as root) [00:00] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_compressed (run as root) [00:01] [FAIL]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_contents (run as root) [00:14] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_deleted (run as root) [00:03] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_disabled_feature (run as root) [00:02] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_embedded (run as root) [00:01] [FAIL]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_holes (run as root) [00:19] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_incrementals (run as root) [00:20] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_largeblocks (run as root) [00:02] [FAIL]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_many_clones (run as root) [00:27] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_mixed_recsize (run as root) [00:06] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_mounts (run as root) [00:03] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_negative (run as root) [00:03] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_origin (run as root) [00:08] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_panic (run as root) [01:51] [PASS]
Test: /home/ryan/zfs/tests/zfs-tests/tests/functional/redacted_send/redacted_props (run as root) [00:09] [PASS]
```
The next test never finishes (and for some reason isn't even being killed).

**Comment by allanjude** [comment]
Should the tests just be updated to run `zstream dump` instead of `zstreamdump`?

**Comment by lundman** [comment]
We've run into this, and somewhat related, we can't run "make install" twice, due to the symlink (zstreamdump -> zstream) already existing (ln -s => ln -fs)

**Comment by behlendorf** [comment]
> we can't run "make install" twice, due to the symlink (zstreamdump -> zstream) already existing (ln -s => ln -fs)

This should be fixed by d484a7255b8fcc31e3602bb89b0b54a994f48d17 which went in a week or so ago.  You might want to cherry-pick it for now, or rebase on the latest master.

**Comment by lundman** [comment]
Cheers, the PR is up to date, but the dev branch is a couple of weeks behind.
